cli/src/main/kotlin/commands/UploadResultToSw360Command.kt

```diff
@@ -81,13 +93,55 @@ class UploadResultToSw360Command : CliktCommand(
         val sw360Connection = Sw360Storage.createConnection(sw360Config, sw360JsonMapper)
         val sw360ReleaseClient = sw360Connection.releaseAdapter
         val sw360ProjectClient = sw360Connection.projectAdapter
+        val downloader = Downloader(globalOptionsForSubcommands.config.downloader)
 
         getProjectWithPackages(ortResult).forEach { (project, pkgList) ->
             val linkedReleases = pkgList.mapNotNull { pkg ->
                 val name = createReleaseName(pkg.id)
-                sw360ReleaseClient.getSparseReleaseByNameAndVersion(name, pkg.id.version)
+                val release = sw360ReleaseClient.getSparseReleaseByNameAndVersion(name, pkg.id.version)
                     .flatMap { sw360ReleaseClient.enrichSparseRelease(it) }
                     .orElseGet { createSw360Release(pkg, sw360ReleaseClient) }
+
+                if (attachSources) {
+                    val tempDirectory = createOrtTempDir(pkg.id.toPath())
+                    try {
+                        // First, download the sources of the package into a sources directory, whose root directory is
+                        // temporary.
+                        val sourcesDirectory = tempDirectory.resolve("sources")
+                        val download = downloader.download(pkg, sourcesDirectory)
+
+                        if (!download.matches(pkg)) {
+                            log.debug { "The sources for the package '${pkg.id.toCoordinates()}' could not be " +
+                                    "uploaded to SW360 due to missing source artifact." }
+                        } else {
+                            // After downloading the source files successfully in a sources directory, create a
+                            // ZIP file of the sources directory and save it in the root directory of it.
+                            // Finally the created ZIP file of the sources can be uploaded to SW360 as an attachment
+                            // of the release.
+                            val zipFile = tempDirectory.resolve("${pkg.id.toPath("-")}.zip")
+                            val archiveResult = archive(sourcesDirectory, zipFile)
+
+                            if (archiveResult.isSuccess) {
+                                val uploadResult = sw360ReleaseClient.uploadAttachments(
+                                    AttachmentUploadRequest.builder(release)
+                                        .addAttachment(zipFile.toPath(), SW360AttachmentType.SOURCE)
```

Comment:
If archiving fails, we currently get no notification at all, as any occurred exception is wrapped into the `Result` object returned by `archive()`. A simple way to solve this is to not check `archiveResult.isSuccess` at all, but to replace `zipFile.toPath()` in this line with `archiveResult.getOrThrow().toPath()`.